auth/env_keychain.go

```diff
@@ -12,27 +12,37 @@ import (
 	"github.com/pkg/errors"
 )
 
-// NewKeychain returns either a EnvKeychain or a authn.DefaultKeychain depending on whether the provided environment variable is set
-func NewKeychain(envVar string) authn.Keychain {
+// ResolveKeychain returns either:
+// * a resolved keychain from the ggcr DefaultKeychain or
+// * a multi-keychain with a resolved keychain from the provided environment variable, and a resolved keychain from the ggcr DefaultKeychain
+// depending on whether the provided environment variable is set
+func ResolveKeychain(envVar string, images []string, overrideKeychain ...authn.Keychain) (authn.Keychain, error) {
```

Comment:
This seems like an awkward way of providing an option argument. From the interface alone I may infer that I can pass multiple keychain and assume that they would be "merged" somehow but that doesn't appear to be the case.

As an example:

```
auth.ResolveKeychain("SOME_VAR", myImage, keychain1, keychain2);
```

Reply:
I think we should refactor this function to take "ops" functions, like [here](https://github.com/buildpacks/lifecycle/blob/63c5c6e6a2f5c6617781ed6d42a80b1c3918730b/testhelpers/docker_cmd.go#L28).

Then when this function is called from `main` it could look like `auth. ResolveKeychain(cmd.EnvRegistryAuth, auth.WithImages(registryImages))` and in the tests it could look like `auth.ResolveKeychain("CNB_REGISTRY_AUTH", auth.WithDefault(fakeKeychain))`.

Reply:
Latest changes should resolve all this weirdness. See https://github.com/buildpacks/lifecycle/pull/409#discussion_r541151115